app/uk/gov/hmrc/exports/repositories/NotificationRepository.scala

```diff
@@ -43,18 +43,29 @@ class NotificationRepository @Inject()(mc: ReactiveMongoComponent)(implicit ec:
     Index(Seq("details" -> IndexType.Ascending), name = Some("detailsDocMissingIdx"), partialFilter = Some(BSONDocument("details" -> BSONNull)))
   )
 
-  def findNotificationsByActionId(actionId: String): Future[Seq[Notification]] =
-    find("actionId" -> JsString(actionId))
+  def findNotificationsByActionId(actionId: String): Future[Seq[ParsedNotification]] =
+    find("actionId" -> JsString(actionId)).map(toParsedNotifications)
 
-  def findNotificationsByActionIds(actionIds: Seq[String]): Future[Seq[Notification]] =
+  def findNotificationsByActionIds(actionIds: Seq[String]): Future[Seq[ParsedNotification]] =
     actionIds match {
       case Seq() => Future.successful(Seq.empty)
-      case _     => find("$or" -> actionIds.map(id => Json.obj("actionId" -> JsString(id))))
+      case _     => find("$or" -> actionIds.map(id => Json.obj("actionId" -> JsString(id)))).map(toParsedNotifications)
     }
 
-  def findUnparsedNotifications(): Future[Seq[Notification]] =
-    find("details" -> JsNull)
+  def findNotificationsByMrn(mrn: String): Future[Seq[ParsedNotification]] =
+    find("details.mrn" -> JsString(mrn)).map(toParsedNotifications)
+
+  def findUnparsedNotifications(): Future[Seq[UnparsedNotification]] =
+    find("details" -> JsNull).map(toUnparsedNotifications)
 
   def removeUnparsedNotificationsForActionId(actionId: String): Future[WriteResult] =
     remove("actionId" -> JsString(actionId), "details" -> JsNull)
+
+  private def toParsedNotifications(notifications: Seq[Notification]): Seq[ParsedNotification] = notifications.collect {
+    case notification: ParsedNotification => notification
+  }
+
+  private def toUnparsedNotifications(notifications: Seq[Notification]): Seq[UnparsedNotification] = notifications.collect {
```

Comment:
Nit: maybe change the names to begin with 'filter' or 'includeOnly' as the name 'toXXXXX' sounds like your are converting them into ParsedNotifications rather than just filtering out the unparsed ones.